x/gamm/twap/logic.go

```diff
@@ -117,19 +121,14 @@ func (k Keeper) getMostRecentRecord(ctx sdk.Context, poolId uint64, assetA, asse
 	if err != nil {
 		return types.TwapRecord{}, err
 	}
-	record = interpolateRecord(record, ctx.BlockTime())
+	record = recordWithUpdatedAccumulators(record, ctx.BlockTime())
 	return record, nil
 }
 
-// interpolate record updates the record's accumulator values and time to the interpolate time.
-//
-// pre-condition: interpolateTime >= record.Time
-func interpolateRecord(record types.TwapRecord, interpolateTime time.Time) types.TwapRecord {
-	return recordWithUpdatedAccumulators(record, interpolateTime)
-}
-
-// TODO: Test math, test p0 vs p1 correctness
-// precondition: endRecord.Time > startRecord.Time
+// precondition: endRecord.Time >= startRecord.Time
+// if (endRecord.Time == startRecord.Time) returns endRecord.LastSpotPrice
+// else returns
+// (endRecord.Accumulator - startRecord.Accumulator) / (endRecord.Time - startRecord.Time)
```

Comment:
Suggesting for the sake of convention, at first I thought this was dead code

```suggestion
// computeArithmeticTwap computes and returns an arithmetic TWAP between
// two records given the quote asset.
// precondition: endRecord.Time >= startRecord.Time
// if (endRecord.Time == startRecord.Time) returns endRecord.LastSpotPrice
// else returns
// (endRecord.Accumulator - startRecord.Accumulator) / (endRecord.Time - startRecord.Time)
```